distribution/docker/Dockerfile

```diff
@@ -51,12 +52,19 @@ RUN --mount=type=cache,target=/root/.m2 VERSION=$(mvn -B -q org.apache.maven.plu
 
 FROM busybox:1.34.1-glibc as busybox
 
-FROM gcr.io/distroless/java$JDK_VERSION-debian11
+FROM $BASE_IMAGE
+ARG JDK_VERSION
+ARG BASE_IMAGE
+
 LABEL maintainer="Apache Druid Developers <dev@druid.apache.org>"
 
 COPY --from=busybox /bin/busybox /busybox/busybox
-RUN ["/busybox/busybox", "--install", "/bin"]
-
+RUN ["/busybox/busybox", "sh", "-c", "if [ $BASE_IMAGE = gcr.io/distroless/java$JDK_VERSION-debian11 ]; then \
+                                        exec /busybox/busybox --install /bin; \
+                                      else \
+                                        rm /busybox/busybox \
+                                        && exit; \
+                                      fi;"]
```

Comment:
Would be nice to add a comment explaining why this is needed.

Reply:
Also, do you need the "&& exit" ? I don't think you should need that.

Reply:
You're right, I'll remove that 👍